tests/pipeline_backend_test.py

```diff
@@ -417,6 +423,43 @@ def tearDownClass(cls):
         cls.sc.stop()
 
 
+class TestLazySingleton(unittest.TestCase):
+
+    def test_with_list(self):
+        ls = LazySingleton([10])
+        self.assertEqual(ls._singleton, 10)
+        self.assertIsNone(ls._iterable)
+        # Also check immediate retrieval
+        self.assertEqual(ls.singleton(), 10)
+
+    def test_init_with_valid_iterable_generator(self):
+        data = (x for x in [10])  # lazy generator
+        ls = LazySingleton(data)
+        self.assertIsNone(ls._singleton)  # Should be lazy
+        self.assertIsInstance(ls._iterable, Iterable)
+        self.assertEqual(ls.singleton(), 10)
+        self.assertEqual(ls.singleton(), 10)
+
+    def test_init_with_empty_list_raises_value_error(self):
+        with self.assertRaisesRegex(ValueError, "exactly one element.*found 0"):
```

Comment:
what `*` means here?

Reply:
result experssion, any number of element